File: PAT/BLevel/1007/1007.cpp
// 1007 素数对猜想
// auguralpha
// 2019-02-11
#include <cstdio>
#include <vector>
using namespace std;
vector<int> vec;

bool if_prime(int n);

int main() {
    int n;
    int cnt = 0;
    scanf("%d", &n);
    for(int i = 2; i <= n; ++i) {
        if(if_prime(i)) {
            vec.push_back(i);
        }
    }
    for(int i = 0; i < vec.size() - 1; ++i) {
        if(vec[i+1] - vec[i] == 2) {
            ++ cnt;
        }
    }

    printf("%d\n", cnt);
    return 0;
}

bool if_prime(int n) {
    for(int i = 2; i * i <= n; ++i) {
        if(n % i == 0) {
            return false;
        }
    }
    return true;
}

File: PAT/BLevel/1007/1007.go
// 1007 素数对猜想
// auguralpha
// 2020-03-25

package main

import "fmt"

func ifPrime(m int) bool {
	for i := 2; i * i <= m; i ++ {
		if m % i == 0 {
			return false
		}
	}
	return true
}

func main() {
	var n int
	var prime []int
	var ans int = 0
	_, _ = fmt.Scanf("%d", &n)
	for i := 1; i <= n; i ++ {
		if ifPrime(i) {
			prime = append(prime, i)
		}
	}

	for i := 0; i < len(prime) - 1; i ++ {
		if prime[i+1] - prime[i] == 2 {
			ans ++
		}
	}

	fmt.Printf("%d\n", ans)
}
